dts/st: enables possibility to define Flash and SRAM out of tree

erwango opened the pull request:
In case of out of tree SoC, mem.h will trigger and error since
it is not aware of new SoC CONFIG_SOC_<reference>.
Modify error condition in order to generate the error only if
the flash and error size are not defined at this point (even in
out of tree case).

Fixes #9978

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

galak commented on dts/arm/st/mem.h:
```diff
@@ -125,7 +125,11 @@
 #define DT_SRAM_SIZE		__SIZE_K(64)
 #else
-#error "Flash, RAM, and CCM sizes not defined for this chip"
+#ifndef DT_FLASH_SIZE
```
can we do this as:
#if !defined(DT_FLASH_SIZE) and !defined(DT_SRAM_SIZE)

galak replied:
Also, wonder if we should put this check at the very top of the file.

galak reviewed (dismissed):
Some minor style feedback.

erwango committed:
dts/st: enables possibility to define Flash and SRAM out of tree

In case of out of tree SoC, mem.h will trigger and error since
it is not aware of new SoC CONFIG_SOC_<reference>.
Modify error condition in order to generate the error only if
the flash and error size are not defined at this point (even in
out of tree case).

Fixes #9978

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
